src/main/java/java/lang/String.java

```diff
@@ -1611,7 +1612,7 @@ public int compareTo(String anotherString) {
     // DIFFBLUE MODEL LIBRARY For some reason this needs to be not null for
     // FileReader tests to pass.
     public static final Comparator<String> CASE_INSENSITIVE_ORDER
-                                            = CProver.nondetWithoutNullForNotModelled();
+                                            = null;
```

Comment:
the comment directly above needs to be updated

Reply:
I don't see why this change is necessary or related to this PR?

Reply:
Some JBMC tests fail because the type on the left and right correspond (for jbmc). This wasn't a problem before but if we load the model of String.java for using String.equals this also gets loaded.